### DISCUSSION TITLE
How to create multiple connect in a single create on many to many relationship ?

### DISCUSSION THREAD

**Discussion by Sakkhor909** [prisma/prisma#11919] [Q&A] [answered]
```prisma
model Category {
  id        String              @id @default(cuid())
  name      String         
  brands    Brand[]
}

model Brand {
  id         String              @id @default(cuid())
  name       String              @unique
  categories Category[]
}
```
Here is my Modal, When users create a Brand, they can insert existing categories or some new categories, The brand should be created new categories or connect to those or existing categories. how can I define it in Prisma Client?

```ts
await prisma.brand.create({
  data: {
    name:  "Samsung",
    categories: ???
  }
});
```

**Comment by austincrim** [comment, marked as answer]
Hey @Sakkhor909 👋,

You will want to take a look at our [`connectOrCreate` API](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connectorcreate). It would look something like this: 

```ts
await prisma.brand.create({
    data: {
      name: 'foo',
      categories: {
        connectOrCreate: {
          create: { name: 'bar' },
          where: { id: 'abc' }
        }
      }
    }
  })
```

Let me know if I can help clarify anything!

**Reply by Sakkhor909** [reply]
Thank you, its work

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
